R/03-annotate.R

```diff
@@ -123,6 +123,12 @@ gimap_annotate <- function(.data = NULL,
     )))
 
   ################################ STORE IT ####################################
+  
+  if (gimap_dataset$filtered_data$filter_step_run){
```

Comment:
Can you explain this chunk? 

Oh is it if filter is run after annotation to filter out annotations that do not apply? 

I suppose its good to allow folks to run it either order. If that's what this is about I think I'm okay with it.

Reply:
If people run filtering before annotation, it'll subset the annotation df. If people run annotation, but not filtering, it'll be the full df. Or at least that's what I was intending/thinking. `gimap_dataset$filtered_data$filter_step_run` is that boolean that I added to check in later steps to see if filtering had been done yet

Reply:
Okay as long as there's some logic to control for whether there's been a filter done or not I think that can work.